From: Mike Rapoport <rppt@kernel.org>
Subject: mm-arch-add-generic-implementation-of-pfn_valid-for-flatmem-fix
Date: Tue, 31 Jan 2023 20:41:49 +0200

The generic pfn_valid() does not take into account pfn_offset when it
compares it with max_mapnr.

Link: https://lkml.kernel.org/r/Y9lg7R1Yd931C+y5@kernel.org
Signed-off-by: Mike Rapoport <rppt@kernel.org>
Reported-by: Conor Dooley <conor.dooley@microchip.com>
Tested-by: Conor Dooley <conor.dooley@microchip.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/include/asm-generic/memory_model.h~mm-arch-add-generic-implementation-of-pfn_valid-for-flatmem-fix
+++ a/include/asm-generic/memory_model.h
@@ -26,7 +26,7 @@ static inline int pfn_valid(unsigned lon
 	extern unsigned long max_mapnr;
 	unsigned long pfn_offset = ARCH_PFN_OFFSET;
 
-	return pfn >= pfn_offset && pfn < max_mapnr;
+	return pfn >= pfn_offset && (pfn - pfn_offset) < max_mapnr;
 }
 #define pfn_valid pfn_valid
 #endif
_
